src/core/counter/tables.rs

```diff
@@ -0,0 +1,380 @@
+            Codex::FirstSeenReplayCouples => "-E", // Composed Base64 Couple, fnu+dts.
+            Codex::TransIdxSigGroups => "-F", // Composed Base64 Group, pre+snu+dig+ControllerIdxSigs group.
+            Codex::SealSourceCouples => "-G", // Composed Base64 couple, snu+dig of given delegators or issuers event
+            Codex::TransLastIdxSigGroups => "-H", // Composed Base64 Group, pre+ControllerIdxSigs group.
+            Codex::SealSourceTriples => "-I", // Composed Base64 triple, pre+snu+dig of anchoring source event
+            Codex::SadPathSig => "-J", // Composed Base64 Group path+TransIdxSigGroup of SAID of content
+            Codex::SadPathSigGroup => "-K", // Composed Base64 Group, root(path)+SaidPathCouples
+            Codex::PathedMaterialQuadlets => "-L", // Composed Grouped Pathed Material Quadlet (4 char each)
+            Codex::AttachedMaterialQuadlets => "-V", // Composed Grouped Attached Material Quadlet (4 char each)
+            Codex::BigAttachedMaterialQuadlets => "-0V", // Composed Grouped Attached Material Quadlet (4 char each)
+            Codex::KERIProtocolStack => "--AAA",         // KERI ACDC Protocol Stack CESR Version
+        }
+    }
+
+    pub(crate) fn from_code(code: &str) -> Result<Self> {
+        Ok(match code {
+            "-A" => Codex::ControllerIdxSigs,
+            "-B" => Codex::WitnessIdxSigs,
+            "-C" => Codex::NonTransReceiptCouples,
+            "-D" => Codex::TransReceiptQuadruples,
+            "-E" => Codex::FirstSeenReplayCouples,
+            "-F" => Codex::TransIdxSigGroups,
+            "-G" => Codex::SealSourceCouples,
+            "-H" => Codex::TransLastIdxSigGroups,
+            "-I" => Codex::SealSourceTriples,
+            "-J" => Codex::SadPathSig,
+            "-K" => Codex::SadPathSigGroup,
+            "-L" => Codex::PathedMaterialQuadlets,
+            "-V" => Codex::AttachedMaterialQuadlets,
+            "-0V" => Codex::BigAttachedMaterialQuadlets,
+            "--AAA" => Codex::KERIProtocolStack,
+            _ => {
+                return Err(Box::new(Error::UnexpectedCode(format!(
+                    "{{\
+                        \"message\":\"unknown code\",\
+                        \"code\":{code}\
+                    }}"
+                ))));
+            }
+        })
+    }
+}
+
+#[cfg(test)]
+mod tables_tests {
+    use super::{hardage, sizage, Codex, Sizage};
+
+    #[test]
+    fn test_hardage() {
+        assert_eq!(hardage("-A").unwrap(), 2);
+        assert_eq!(hardage("-G").unwrap(), 2);
+        assert_eq!(hardage("-V").unwrap(), 2);
+        assert_eq!(hardage("-0").unwrap(), 3);
+        assert_eq!(hardage("--").unwrap(), 5);
+    }
+
+    #[test]
+    fn test_sizage() {
+        let mut s: Sizage;
+
+        s = sizage("-A").unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 2);
+        assert_eq!(s.fs, 4);
+        assert_eq!(s.ls, 0);
```

Comment:
Here is how I generated this (irb/ruby):

```
3.0.0 :001 > s.lines.each do |line|
3.0.0 :002 >   line =~ /\s+'(.+)': Sizage\(hs=(\d), ss=(\d), fs=(\d), ls=(\d)\)/
3.0.0 :003 >   puts "s = sizage(\"#{$1}\");"
3.0.0 :004 >   puts "assert_eq!(s.hs, #{$2});"
3.0.0 :005 >   puts "assert_eq!(s.ss, #{$3});"
3.0.0 :006 >   puts "assert_eq!(s.fs, #{$4});"
3.0.0 :007 >   puts "assert_eq!(s.ls, 0);"
3.0.0 :008 >   puts
3.0.0 :009 > end
```

Reply:
from this input: https://github.com/WebOfTrust/keripy/blob/development/src/keri/core/coring.py#L4282-L4296